execute_impf/execute_impf.rst

```diff
@@ -0,0 +1,1033 @@
+                                  "donutSelector.fluxField": 'g_flux',                            
+                            }
+                        }
+                    }
+                }
+    )
+
+Before executing the command, let’s clear the events we will want to
+check afterwards. This is mostly to make sure there are no previous
+events in the queue that could affect the verification process
+afterward.
+
+.. code:: ipython3
+
+    mtcs.rem.mtaos.evt_wavefrontError.flush()
+    mtcs.rem.mtaos.evt_wepDuration.flush()
+
+.. code:: ipython3
+
+    start_time = utils.current_tai()
+    await mtcs.rem.mtaos.cmd_runWEP.set_start(
+        visitId=4021123110021,
+        config=wep_config,
+    )
+    end_time = utils.current_tai()
+
+.. code:: ipython3
+
+    print(f"Execution took {end_time-start_time}s")
+
+
+.. parsed-literal::
+
+    Execution took 289.32793498039246s
+
+
+--------------
+
+Verify output
+~~~~~~~~~~~~~
+
+To verify the execution we will check the ``wepDuration`` and
+``wavefrontError``. The first one contains the measured execution time
+done by the CSC whereas the second contains a set events with the result
+of the wavefront estimation pipeline.
+
+The CSC publishes one sample of ``wavefrontError`` for each pair of
+donuts processed, so there are a number of events published.
+
+.. code:: ipython3
+
+    print(await mtcs.rem.mtaos.evt_wepDuration.next(flush=False, timeout=STD_TIMEOUT))
+
+
+.. parsed-literal::
+
+    private_revCode: 20fbb051, private_sndStamp: 1646437733.040275, private_rcvStamp: 1646437733.04058, private_seqNum: 7, private_identity: MTAOS, private_origin: 38497, calcTime: 290.7162170410156, priority: 0
+
+
+--------------
+
+The next cell will loop over the events and save the results to a list
+so we can plot them afterwards.
+
+.. code:: ipython3
+
+    wavefront_errors = []
+    
+    while True:
+        try:
+            wfe = await mtcs.rem.mtaos.evt_wavefrontError.next(flush=False, timeout=1)
+            wavefront_errors.append(wfe)
+        except Exception:
+            break
+
+.. code:: ipython3
+
+    for wfe in wavefront_errors:
+        plt.plot(wfe.annularZernikeCoeff)
+
+
+
+.. image:: output_39_0.png
+
+
+.. code:: ipython3
+
+    print(wavefront_errors[0])
+
+
+.. parsed-literal::
+
+    private_revCode: 7a035a53, private_sndStamp: 1646437733.0358386, private_rcvStamp: 1646437733.0363667, private_seqNum: 73, private_identity: MTAOS, private_origin: 38497, sensorId: 191, annularZernikeCoeff: [-0.6501623898635709, 0.3798855746064728, 0.4171688832702315, -0.07065003643697557, 0.31799444845953906, -0.09458405122403776, -0.28967372843448186, -0.009719830879297234, 0.010083113576603074, -0.00658564906902043, -0.10764683917981688, -0.002811593258348531, -0.08759269476783502, 0.053523393682423905, -0.022526749693036415, -0.015325838641795759, 0.08600518092616989, -0.05175709824344097, 0.013962613446118372], priority: 0
+
+
+.. _run-ofc:
+
+Run OFC
+-------
+
+…
```

Comment:
These `...` called my attention in the sense that probably we want some minor description here? Having a deeper look, I believe that adding short descriptions to each of the plots might help other people to understand them. And I believe that one of the questions I would have is "what am I expected to see in these plots?" or "how do I know I am getting reasonable results?"

Reply:
I added some information about the Run OFC command. 

The second part of your question is more complicated. I think it is a bit outside the scope of this exercise to dive into the details of debugging the results of WEP and OFC. The main reason is that it would require a deep dive into the details of the system is order to accomplish this task.

I'll leave a comment at the top of the notebook with some links to where people can go to get more information about it, but I don't want to make this an "AOS user manual"...